text/0057-single-field-shortcut.md

```diff
@@ -0,0 +1,167 @@
+                m.d.sync += self.toggle[i].eq(0)
+
+        return m
+```
+
+The `toggle` name is used, among other ways, to name the register class, to name
+the field within, and to access that field on the register instance during
+elaboration.
+
+This can be simplified by passing the `csr.Field` directly to the `csr.Register`
+without enclosing it in a dict/list and naming/indexing it, or subclassing
+`csr.Register`. This also causes the `.f` member to access that Field directly
+instead of needing to provide a name/index.
+
+This simplifies the design as follows, reducing clutter and redundancy:
+
+```python
+from amaranth import *
+from amaranth.lib import wiring
+from amaranth.lib.wiring import In, Out
+from amaranth_soc import csr
+
+class Toggle(wiring.Component):
+    bus: In(csr.Signature(addr_width=1, data_width=32))
+    toggle: Out(32)
+
+    def __init__(self):
+        # Register can take the Field directly, eliminating the subclass
+        self._toggle = csr.Register(csr.Field(csr.action.RW, 32), access="rw")
+
+        # ... bridge setup ...
+
+    def elaborate(self, platform):
+        m = Module()
+
+        # ... bridge setup ...
+
+        for i in range(32):
+            # FieldAction accessed directly, no need to specify `toggle` again
+            with m.If(self._toggle.f.data[i]):
+                m.d.sync += self.toggle[i].eq(~self.toggle[i])
+            with m.Else():
+                m.d.sync += self.toggle[i].eq(0)
+
+        return m
+```
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+(with reference to [RFC #16](https://github.com/amaranth-lang/rfcs/blob/main/text/0016-soc-csr-regs.md))
+
+#### `csr.reg.Register`
+
+The `csr.reg.Register` class describes a CSR register `Component`, with:
+- a `.__init__(self, fields=None, access=None)` constructor, where:
+    * `fields` is either:
+        - a `dict` that will be instantiated as a `FieldActionMap`;
+        - a `list` that will be instantiated as a `FieldActionArray`;
+        - a `Field` that will be instantiated as a `FieldAction`;
+        - `None`; in this case a `FieldActionMap` is instantiated from `Field` objects in variable annotations.
+    * `access` is either a `csr.Element.Access` value, or `None`.
+- a `.fields` property, returning the instantiated `FieldActionMap`/`FieldActionArray`/`FieldAction`;
+- a `.f` property, as a shorthand to `self.fields`;
+- a `.__iter__(self)` method that yields, for each contained field, a tuple containing its path (as a tuple of names or indices) and its instantiated `FieldAction`. If only a single `Field` was supplied, its path is always `tuple()`.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+* A novice might not understand that it's possible to create a `Register` with multiple `Fields` if exposed only to code which uses single-`Field` `Register`s.
+* There is a difference between a `Register` with one named/indexed `Field` and with one unnamed `Field`.
```

Comment:
This seems fine since the former suggests the register may be expanded in the future but the latter suggests the register will stay exactly as it is.